lib/pbench/server/api/resources/datasets_compare.py

```diff
@@ -21,14 +21,43 @@
     Schema,
 )
 from pbench.server.cache_manager import CacheManager
-from pbench.server.database.models.datasets import Metadata
+from pbench.server.database.models.datasets import Dataset, Metadata
 
 
 class DatasetsCompare(ApiBase):
     """
     This class implements the Server API used to retrieve comparison data for visualization.
     """
 
+    @staticmethod
+    def get_benchmark_name(dataset: Dataset) -> str:
+        """Convenience function to get dataset's benchmark
+
+        The Pbench Server intake constructs a server.benchmark metadata key to
+        represent the benchmark type. This helper implements a fallback for
+        datasets processed prior to the implementation of server.benchmark to
+        avoid null values, by using the Pbench Agent metadata.log "script"
+        value if that exists and then a constant fallback value.
+
+        TODO: This is a workaround from the transition to server.benchmark in
+        order to cleanly support earlier datasets on the staging server. This
+        can be removed at some point, but it's not critical.
+
+        Args:
+            dataset: Dataset object
+
+        Returns:
+            A lowercase benchmark identifer string, including the value defined
+            by Metadata.SERVER_BENCHMARK_UNKNOWN if we can't find a value.
+        """
+        benchmark = Metadata.getvalue(dataset, Metadata.SERVER_BENCHMARK)
+
+        if not benchmark:
```

Comment:
If you edit this for some other reason, I would remove this blank line (e.g., you didn't put one after the other call to `getvalue()`...).